ghproxy/ghmetrics/ghmetrics.go

```diff
@@ -0,0 +1,61 @@
+package ghmetrics
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+// ghTokenUsageGaugeVec provides the 'github_token_usage' gauge that
+// enables keeping track of GitHub calls and quotas.
+var ghTokenUsageGaugeVec = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "github_token_usage",
```

Comment:
Don't we need to partition this based on token?

Reply:
How would we get the token info at this point? Do you want to write a hash of the sent token collected from the request header? Do we have info on the token name used somewhere at hand?

Reply:
I think our only way of doing it would be to cryptographically hash the authorization header, yeah.

Reply:
I will look into this tomorrow.